pkg/asset/installconfig/gcp/gcp.go

```diff
@@ -82,13 +81,38 @@ func selectProject(ctx context.Context) (string, error) {
 	return selectedProject, err
 }
 
-func selectRegion(project string) (string, error) {
-	longRegions := make([]string, 0, len(validation.Regions))
-	shortRegions := make([]string, 0, len(validation.Regions))
-	for id, location := range validation.Regions {
-		longRegions = append(longRegions, fmt.Sprintf("%s (%s)", id, location))
-		shortRegions = append(shortRegions, id)
+func selectRegion(ctx context.Context, project string) (string, error) {
+	ssn, err := GetSession(ctx)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get session")
 	}
+
+	client := &Client{
+		ssn: ssn,
+	}
+
+	selectRegionCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+
+	validRegions, err := client.GetRegions(selectRegionCtx, project)
+	if err != nil {
+		return "", errors.Wrap(err, "failed to get regions")
```

Comment:
These are the regions that would be included in the survey, so it makes sense to filter here based on the list from pkg types.